spec/requests/filter_parameter_logging_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+
+RSpec.describe "Filter parameter logging spec", type: :request do
+  let(:request_user) { FactoryBot.create(:user) }
+  let(:sync_route) { "/api/v3/patients/sync" }
+  let(:build_payload) { -> { build_patient_payload(FactoryBot.build(:patient, registration_facility: request_user.facility)) } }
+
+  let(:auth_headers) do
+    {"HTTP_X_USER_ID" => request_user.id,
+     "HTTP_X_FACILITY_ID" => request_user.facility.id,
+     "HTTP_AUTHORIZATION" => "Bearer #{request_user.access_token}"}
+  end
+  let(:headers) do
+    {"ACCEPT" => "application/json", "CONTENT_TYPE" => "application/json"}.merge(auth_headers)
+  end
+
+  it "does not log sensitive patient data" do
+    stringio = StringIO.new
+    test_logger = Logger.new(stringio)
+    allow(ActionController::Base).to receive(:logger).and_return(test_logger)
+
+    patient_payloads = (1..3).map { build_payload.call }
```

Comment:
I've always written this as `3.times.map { build_payload.call }` so it's one less thing to track/break